tests/flox-init.bats

```diff
@@ -0,0 +1,232 @@
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test `flox init' sub-command.
+#
+# This sub-command's `--template' option takes an "installable" URI as its
+# argument, and we focus on auditing various URI edge cases here.
+#
+# NOTE: the URIs accepted by `flox init --template <URI>' do not accept an
+# extended output spec ( `^out,bin,dev' suffix ).
+#
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash;
+
+# bats file_tags=uri, init, uri:project
+
+
+# ---------------------------------------------------------------------------- #
+
+setup_file() {
+  common_file_setup;
+
+  # Suppresses warning messages that clutter backtraces.
+  git config --global init.defaultBranch main;
+
+  export _floxpkgs_rev="2c75b96bc3e8c78b516b1fc44dbf95deae6affca";
```

Comment:
Is the idea here to make sure that tests always run deterministically? I think that's a good idea in terms of having a deterministic test suite, but I also think it's probably good for us to be testing against current packages. Could you loop over this revision and something close to "latest"?

Reply:
No the purpose is to use a valid revision number to audit the parser.
The revision is arbitrary as long as it is a valid one which has the `project` template with `shell/__PACKAGE_NAME__/default.nix` in it.